Gemfile.lock

```diff
@@ -170,17 +161,16 @@ PLATFORMS
 DEPENDENCIES
   bourbon
   builder (~> 3.0)
-  foreman
+  bundler (~> 1.17.3)
```

Comment:
There is a version of middleman that supports bundler 2

Reply:
I wasn't prepared for cleanup & full update, but if you ask for it, sure :D

Reply:
After the update to Middleman 4:
```
/Users/pirj/.rvm/gems/ruby-2.5.1@rspec.github.io/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:96:in `load': Tried to activate old-style extension: deploy. They are no longer supported. (RuntimeError)
```

I don't see a clear way to deploy otherwise, might be `git push staging master` would work. 
Also, `source_dir` in `RSpecInfo::Helpers.rspec_documentation_latest(source_dir)` is not defined, so probably version fallback is broken.

I'd say there's a lot of work with no tangible benefit. Let's stick with Bundler 1 for now?

Reply:
@JonRowe Can you please grant me permissions to push to https://github.com/rspec-staging/rspec-staging.github.io/ 's `master`?

I intend to rework the deployment in the following way:
 - update `middleman` and `middleman-core` to the latest
 - drop `middleman-deploy` and its configs
 - use Bundler 2
 - make `build` directory a submodule of this very repository (and will make a note to avoid `git initialize --recursive` just in case, even though the submodule will be on `master` with no submodules defined)
 - add info how to add two remotes to that submodule, itself and rspec-staging
 - to publish: prod `(cd build; git push prod master)`, staging `(cd build; git push staging master)`
 - forget about the staging trick of omitting `CNAME`, if a "Custom Domain" is not filled in in `rspec-staging/rspec-staging.github.io/`, it should not redirect to the domain then

I'd love to test how this will work on staging

Reply:
Done, but bare in mind that I'm not sure submodules work with github pages...

Reply:
It looks like middleman 4.0 update is a major PITA.
The documentation works quite ok
I'm stuck with an endless list of SASS failures:
```
== Request: /stylesheets/_application.css
       error  build/stylesheets/_application.css
Error: File to import not found or unreadable: bourbon.
        on line 5:1 of source/stylesheets/_application.css.scss
>> @import "bourbon";

   ^

...

== Request: /stylesheets/_application.css
   identical  build/stylesheets/_code.css
       error  build/stylesheets/_application.css
Error: File to import not found or unreadable: bourbon.
        on line 5:1 of source/stylesheets/_application.css.scss
>> @import "bourbon";

   ^

/Users/pirj/source/rspec-dev/repos/rspec.github.io/source/stylesheets/_application.css.scss:5
```

Besides that, I keep finding new things I'd love to improve, namely ditch `neat` which is not maintained anymore, and it's been recommended to use flexbox instead.

Even with the current dependencies we're getting quite a lot of:
```
WARNING: The button mixin is deprecated and will be removed in the next major version release
```
I don't really feel like fixing all of that at once.

Good to merge as is and address middleman 4/neat removal/css fixes separately?